### PR TITLE
Added install fix for pyldap

### PR DESCRIPTION
**Pull request by gruunday**

### CODE REVIEW

**Comment by butlerx** [comment]
you shouldnt need this as you are using the python3.5 docker container

**Comment by butlerx** [comment]
you shouldnt need this as you are using the python3.5 docker

**Comment by butlerx** [comment]
you shouldnt need this as you are using the python3.5 docker

**Comment by gruunday** [comment]
Your install pyldap is failing. These are trying to fix that

**Comment by butlerx** [comment]
you was to add `libldap2-dev` and `libsasl2-dev`

**Comment by gruunday** [comment]
https://github.com/pyldap/pyldap/blob/master/INSTALL

Thought could be causing the error

**Comment by butlerx** [comment]
its just missing some header files those packages should cover

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,10 +4,13 @@ jobs:
     docker:
       - image: python:3.5
     working_directory: /home/ubuntu/useradm
+    dependencies:
+      pre:
+        - sudo apt-get update; sudo apt-get install
```
Comment (butlerx) [inline .circleci/config.yml]: `apt-get update && sudo apt-get install libldap2-dev libsasl2-dev`

File: .circleci/config.yml [hunk 2]
```diff
@@ -4,10 +4,13 @@ jobs:
...
     steps:
       - checkout
-      - run: pip install -r requirements.txt
-      - run: pip install pylint
+      - run: pip3 install -r requirements.txt
```
Comment (butlerx) [inline .circleci/config.yml]: leave them as pip as python 3 is the default in the container

**Review by butlerx — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (gruunday) [reply .circleci/config.yml]: Trying to rule out things why requirements won't install. If I fix all these changes we have no PR. I don't plan to merge this. At this point it is just Debugging what is wrong

Comment (butlerx) [reply .circleci/config.yml]: did you try installing the libraries i said you needed

Comment (gruunday) [reply .circleci/config.yml]: what libraries did you say to install?

Comment (butlerx) [reply .circleci/config.yml]: see comment appove

Comment (gruunday) [reply .circleci/config.yml]: I see it. I read it. I still don't understand it.

Comment (gruunday) [reply .circleci/config.yml]: pip and pip3 give different error messages. I feel that is the confusion

File: .circleci/config.yml [hunk 3]
```diff
@@ -4,10 +4,14 @@ jobs:
     docker:
       - image: python:3.5
     working_directory: /home/ubuntu/useradm
+    dependencies:
+      pre:
+        - sudo apt-get update; sudo apt-get install gcc-5; sudo apt-get install g++-5
```
Comment (butlerx) [inline .circleci/config.yml]: these shoudl not be needed

File: .circleci/config.yml (hunk 2 above)
Comment (butlerx) [reply .circleci/config.yml]: they shoudlnt as pip and pip3 are the same

File: .circleci/config.yml [hunk 4]
```diff
@@ -4,10 +4,14 @@ jobs:
...
+        - sudo apt-get install libsasl2-dev python3-dev libldap2-dev libssl-dev
```
Comment (butlerx) [inline .circleci/config.yml]: python3-dev shoudl already be installed

File: .circleci/config.yml [hunk 5]
```diff
@@ -4,10 +4,14 @@ jobs:
     docker:
       - image: python:3.5
     working_directory: /home/ubuntu/useradm
+    dependencies:
```
Comment (butlerx) [inline .circleci/config.yml]: your biggest problem is this is wrong these shoudl be steps

File: .circleci/config.yml [hunk 6]
```diff
@@ -6,8 +6,9 @@ jobs:
     working_directory: /home/ubuntu/useradm
     steps:
       - checkout
-      - run: pip install -r requirements.txt
-      - run: pip install pylint
+      - run: apt-get update && apt-get install libldap2-dev libsasl2-dev
```
Comment (butlerx) [inline .circleci/config.yml]: add `-y` too `apt install`

File: .circleci/config.yml [hunk 7]
```diff
@@ -6,8 +6,9 @@ jobs:
...
+      - run: pip3 install -r requirements.txt
```
Comment (butlerx) [inline .circleci/config.yml]: best to switch back to pip

**Review by butlerx — COMMENTED** [review]
you were using a mix of circle v1 and v2 syntax we use v2
ive fixed it to version2

**Comment by gruunday** [comment]
We fixed it. Was going to open new request rather than this mess go into the history

**Comment by butlerx** [comment]
itll get squashed in to one cmt

**Review by butlerx — APPROVED** [review]

**Pull request merged**